tests/conftest.py

```diff
@@ -990,7 +989,8 @@ def postgis_db():
         try:
             conn.execute("SELECT postgis_version();")
         except sqlalchemy.exc.DBAPIError:
-            raise pytest.skip("Requires PostGIS")
+            pass
+            # raise pytest.skip("Requires PostGIS")
```

Comment:
This fixture's called `postgis_db`, so I think marking a test with it should ensure we have postgis available. Probably we need to split this into two fixtures called (`postgres_db`, `postgis_db`)

`postgis_db` will require `postgres_db`, and additionally check that postgis is installed. Your new test should use `postgres_db` rather than hardcoding db settings itself

Reply:
FYI what you see here is the hack I helped Elena write that allows her to develop the test `test_import_from_non_postgis_db` but I didn't settle on any particular plan going forward - if we need to env flags eg KART_POSTGRES_URL and KART_POSTGIS_URL pointing to two different PG instances, if we need to spin up both during CI. That was basically the only plan I thought of - that or not running this test very often since it requires a special environment - and it seemed a bit heavyweight for a single test, probably there are other options though.

Reply:
The other way to do this is to have a third fixture (`postgres_db`, `postgis_db`, and `no_postgis_db`). The latter depends on `postgres_db` and just runs `DROP EXTENSION IF EXISTS postgis`, within a transaction, and rolls it back afterwards.


It seems like you can actually do that (yay postgres!):

```
x=# begin;
BEGIN
x=*# drop extension postgis;
DROP EXTENSION
x=*# select st_asewkt('point(0 0)'::geometry);
ERROR:  type "geometry" does not exist
LINE 1: select st_asewkt('point(0 0)'::geometry);
                                       ^
x=!# rollback;
ROLLBACK
x=# select st_asewkt('point(0 0)'::geometry);
 st_asewkt
------------
 POINT(0 0)
(1 row)
```

However, for this to work we do need superuser access to the database.

Reply:
what about now?

Reply:
looks great 👍